cdk/lib/iot-application-stack.ts

```diff
@@ -13,51 +19,94 @@ export class IotApplicationStack extends Stack {
   constructor(scope: Construct, id: string, props: LoggingStackProps) {
     super(scope, id, props);
 
+    const authMode = this.node.tryGetContext('authMode') as string;
+
     const {
       logGroup: { logGroupArn },
     } = new LoggingStack(this, 'Logging', {
       ...props,
       applicationName: id,
     });
 
-    const {
-      userPool: { userPoolId },
-      userPoolClient: { userPoolClientId },
-      identityPool: { ref: identityPoolId },
-    } = new AuthStack(this, 'Auth', {
-      ...props,
-      applicationName: id,
-      logGroupArn,
-    });
-
     const {
       resourceTable: { tableArn, tableName },
     } = new DatabaseStack(this, 'Database', props);
 
-    const {
-      coreService: {
-        service: { attrServiceUrl: coreServiceUrl },
-      },
-    } = new CoreStack(this, 'Core', {
-      ...props,
-      coreServiceProps: {
+    // Cognito domain is required for SSO
+    if (authMode === AuthModeOptions.SSO) {
+      const {
+        userPool: { userPoolId },
+        userPoolClient: { userPoolClientId },
+        identityPool: { ref: identityPoolId },
+        domain: { domainName },
+      } = new SsoAuthStack(this, 'Auth', {
+        ...props,
         applicationName: id,
-        databaseTableArn: tableArn,
-        databaseTableName: tableName,
-        identityPoolId: identityPoolId,
-        userPoolClientId: userPoolClientId,
-        userPoolId: userPoolId,
-      },
-    });
+        logGroupArn,
+      });
 
-    new CfnOutput(this, 'AppURL', {
-      description: 'Endpoint to access the App',
-      value: `https://${coreServiceUrl}`,
-    });
+      const {
+        coreService: {
+          service: { attrServiceUrl: coreServiceUrl },
+        },
+      } = new CoreStack(this, 'Core', {
+        ...props,
+        coreServiceProps: {
+          applicationName: id,
+          databaseTableArn: tableArn,
+          databaseTableName: tableName,
+          identityPoolId: identityPoolId || '',
+          userPoolClientId: userPoolClientId || '',
```

Comment:
> [!IMPORTANT]  
[type safety] when are those values be falsey? the application doesn't work when they are `''` 🤔

Reply:
I think those were remnants from before restructuring it, removed in latest push.